### DISCUSSION TITLE
Custom logic to expose thrown errors from resolvers?

### DISCUSSION THREAD

**Discussion by m1212e** [hayes/pothos#1334] [Q&A] [answered]
I'm trying to expose the prisma errors with the codes `P2025` and `P2002` which in my app can safely be exposed to the client.

```ts
if (error.code === 'P2025' || error.code === 'P2002') {
    return error.message;
}
```

I know about 
```ts
throw new GraphQLError('my cool custom message');
```
which works for me (using yoga).

But I don't want to wrap every db call into a try catch block which maps the errors, I would like to do this globally. Something like

Error is thrown => My function runs, decides to either throw a GraphQLError or simply return the thrown error => Error gets returned

Is something like this possible?
Thanks!

**Comment by hayes** [comment, marked as answer]
I think you are looking for this: https://the-guild.dev/graphql/yoga-server/docs/features/error-masking#customize-error-masking

**Reply by edwinzhng** [reply]
Does anyone have an example implementation of this anywhere? I'm running Pothos + Yoga and when I try to catch a `Prisma. PrismaClientKnownRequestError`, the error I receive in my `maskedErrors` function from the above link only contains the stack trace, but drops the `code` and `meta` fields that are typically present on `PrismaClientKnownRequestError`, which makes it difficult to determine which error to catch
